### PR TITLE
fix(Input): fix size prop (#660)

### PR DESCRIPTION
**Pull request by gergely-nagy**

### CODE REVIEW

File: src/Input.js [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import { mapToCssModules, deprecated } from './utils';
+import { mapToCssModules, deprecated, warnOnce } from './utils';
 
 const propTypes = {
   children: PropTypes.node,
   type: PropTypes.string,
   size: PropTypes.string,
+  bsSize: PropTypes.oneOf(['lg', 'sm']),
```
Comment (TheSharpieOne) [inline src/Input.js]: Don't limit this. If someone wants to extend bootstrap to add something like xl (or anything else) they should be able to just add their class and pass it here.

File: src/Input.js [hunk 2]
```diff
@@ -72,11 +73,16 @@ class Input extends React.Component {
       }
     }
```
Comment (TheSharpieOne) [inline src/Input.js]: I'd do a regex check to see if it is a word/not a number to trigger this logic. It works with the custom sizes better.

File: src/Input.js [hunk 3]
```diff
@@ -72,11 +73,16 @@ class Input extends React.Component {
...
 
+    if (['lg', 'sm'].indexOf(attributes.size) > -1) {
```
Comment (TheSharpieOne) [inline src/Input.js]: You should also delete size from attributes so it doesn't get passed down to the DOM in this case.

File: src/Input.js [hunk 4]
```diff
@@ -43,6 +44,7 @@ class Input extends React.Component {
     } = this.props;
 
     const checkInput = ['radio', 'checkbox'].indexOf(type) > -1;
+    const isNotaNumber = new RegExp('(?!^\\d+$)^.+$');
```
Comment (TheSharpieOne) [inline src/Input.js]: `/\D/g` would work and is more clear. `new RegExp('\\D','g')` if you want to use `new RegExp`

Comment (gergely-nagy) [reply src/Input.js]: Yes , this is much better. Sorry, i forgot this global flag . Thank you for taking your time for the review. I'll fix it right away.

**Pull request merged**